### ISSUE TITLE
Provide guidance to Azure users on setting up Teleport backends

### ISSUE THREAD

**Issue by anthonysomerset** [gravitational/teleport#22072]
## Applies To
https://goteleport.com/docs/reference/backends/

## Details
The Storage backends mentions that all the Data Types can use the file system as a storage backend, and does suggest for HA to use the likes of DynamoDB and S3 storage.

We are heavy Azure users and don't really use AWS so we are in a catch 22 as we don't want to use AWS just for initial testing and then later deployment.

Azure Kubernetes Service can mount persistent volumes based in Azure File (using SMB) or Azure Blob (using NFS), inside the actual pods they just appear as directories like any other mounted network filesystem and can be interacted with in exactly the same way.

It would be good if its possible to use a shared file storage backend like this for HA - at least until the likes of #15147 are fully released, right now we'd even be happy with it even if its only considered "experimental" as we look to evaluate the solution more thoroughly.

Apologies i could not find an existing case that speaks to this - only perhaps only tangentially - #11409

If this is possible then it would be good if the documentation could be updated to speak to this scenario accordingly - this is a good case of the phrase "You can "outsource" this function to the infrastructure." can apply

I would be comfortable with a scenario where the core cluster state can't use shared FS so must use something like etcd but audit and session events could use a shared FS accordingly - as much as i'd prefer to not maintain an etcd cluster either :)

**Comment by anthonysomerset** [comment]
I took something of a first pass on this

Cluster state in etcd
events and session as an Azure Blob NFS mount exposed as a directory `/datastore/` to the pods, and then config to reference folders under this (that exist)

The pods can see the NFS mount just fine, if i enter into the pod, ls chmod etc work fine

Sessions appear to be uploaded fine also - I see them on the blob side and after a brief delay they show in the gui also...

Audit Log appears in the GUI but does not appear to be uploaded to blob storage - i see the directory is empty on disk as well as on blob storage - as a result i think the gui will only see one half of audit logs and will reset on pod reboot etc

no obvious errors in the logs except perhaps

```
2023-02-22T14:26:17Z [AUDIT]     WARN Failed to parse audit log file "events.log" format: parsing time "events" as "2006-01-02.15:04:05": cannot parse "events" as "2006" events/filelog.go:503
```

**Comment by anthonysomerset** [comment]
OK so i have what i think is an understanding as to why audit logs cannot use shared storage...

by default they will log to same path/file, so therefore if you log from multiple auth nodes, they can't all write to the same file at same time, so i think it gets into a race condition.

The only way to use a shared storage medium would be for the auth nodes to write to a unique file per node e.g. <hostname>-events.log but thats probably a separate discussion.

**Comment by zmb3** [comment]
Postgres and Azure Blob storage backends were released in 13.3
